### PR TITLE
feat(core): Add commands to encrypt and decrypt nano tdfs

### PR DESCRIPTION
**Pull request by elizabethhealy**
Follow up to go sdk nano work

### CODE REVIEW

File: pkg/handlers/nano-tdf.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		),
+	)
+	if err != nil {
+		return nil, err
+	}
+	return enc, nil
+}
+
+func (h Handler) DecryptNanoTDF(toDecrypt []byte) (*bytes.Buffer, error) {
+	tdfreader, err := h.sdk.LoadNanoTDF(bytes.NewReader(toDecrypt))
```
Comment (elizabethhealy) [inline pkg/handlers/nano-tdf.go:31]: TBD not sure of the api here, dont know what the function name will be

File: pkg/handlers/nano-tdf.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+	"github.com/opentdf/platform/sdk"
+)
+
+func (h Handler) EncryptNanoBytes(b []byte, values []string) (*bytes.Buffer, error) {
+	var encrypted []byte
+	enc := bytes.NewBuffer(encrypted)
+
+	// TODO: validate values are FQNs or return an error [https://github.com/opentdf/platform/issues/515]
+	_, err := h.sdk.CreateNanoTDF(enc, bytes.NewReader(b),
```
Comment (elizabethhealy) [inline pkg/handlers/nano-tdf.go:16]: TBD not sure of the api here, dont know what the function name will be

File: go.work [hunk 3]
```diff
@@ -0,0 +1,6 @@
+go 1.22.3
+
+use (
+	.
+	../opentdf-v2-poc/lib/ocrypto
+)
```
Comment (elizabethhealy) [inline go.work:6]: REMOVE once sdk is published

File: pkg/handlers/tdf.go [hunk 4]
```diff
@@ -16,7 +15,7 @@ func (h Handler) EncryptBytes(b []byte, values []string) (*bytes.Buffer, error)
 	_, err := h.sdk.CreateTDF(enc, bytes.NewReader(b),
 		sdk.WithDataAttributes(values...),
 		sdk.WithKasInformation(sdk.KASInfo{
-			URL:       fmt.Sprintf("http://%s", h.platformEndpoint),
+			URL:       h.platformEndpoint,
```
Comment (elizabethhealy) [inline pkg/handlers/tdf.go:18]: the --host flag expected a url with a scheme already, dont need to add it here

**Comment by elizabethhealy** [comment]
roundtrip working:
```
go run . auth client-credentials -i <client> -s <secret> 
echo "helloworld" | go run . encrypt-nano -o sensitive.txt.tdf --host http://localhost:8080 --tls-no-verify --log-level debug
go run . decrypt-nano sensitive.txt.tdf --host http://localhost:8080 --tls-no-verify --log-level debug
```

**Comment by jakedoublev** [comment]
Thanks for this @elizabethhealy. Would it be more elegant to drive the existing `encrypt/decrypt` CLI subcommands based on a new `--tdf-type` flag with a default of TDF3 (I think) for the existing functionality, and this new NanoTDF functionality run if we have `--tdf-type nano`?

It seems like most of this stdin/stdout/file handling is repeated and we could avoid 2 more subcommands. 🤔

**Comment by elizabethhealy** [comment]
> Thanks for this @elizabethhealy. Would it be more elegant to drive the existing `encrypt/decrypt` CLI subcommands based on a new `--tdf-type` flag with a default of TDF3 (I think) for the existing functionality, and this new NanoTDF functionality run if we have `--tdf-type nano`?
> 
> It seems like most of this stdin/stdout/file handling is repeated and we could avoid 2 more subcommands. 🤔

@jakedoublev ya i agree that's much cleaner

**Review by jakedoublev — APPROVED** [review]

**Pull request merged**